BuildingAndTestingInBunsen.md

```diff
@@ -0,0 +1,119 @@
+
+* [Set up a Bunsen Environment](Getting_Started.md)
+
+# Expectations of this Guide
+
+This guide is currently intended to provide enough information to get minimally functional.  A more detailed guide will be generated later, but in the meantime this document should help you get started if you're interested in contributing to the VDO projects.
+
+# Find the appropriate code repository
+
+Identify which repository you want to test.
+
+Currently the following VDO repositories are set up to allow for local testing:
+* [vdo-devel](https://github.com/dm-vdo/vdo-devel) - This is where the VDO product code resides as well as any supporting test code that is only needed for VDO itself.
+* [common](https://github.com/dm-vdo/common) - This is where the library code resides that supports both the test environment as well as VDO.
+
+* How do I tell what's the difference between the two?
+  * The first thing to figure out is whether you're modifying the VDO code itself or the tests that directly test VDO.
+    * If that's the case, then you want to use the [vdo-devel](https://github.com/dm-vdo/vdo-devel) tree.
+  * Otherwise, you'll need to look for the code in either repository.  If you're not sure, then you should likely start on the [vdo-devel](https://github.com/dm-vdo/vdo-devel) repository and move to the [common](https://github.com/dm-vdo/common) repository if you still couldn't find what you were looking for.
+
+# Find the code
+
+For the purposes of this guide, all of the work you will be doing is going to take place on the Bunsen `resource` system.
+
+SSH to the `resource` system (typically named `ossbunsen-resource`) as the user that you have provisioned it with.
+
+
+For changes to the [vdo-devel](https://github.com/dm-vdo/vdo-devel) repository:
+
+  * Starting in the home directory, clone the repository and cd into it
+    ```
+    git clone https://github.com/dm-vdo/vdo-devel.git && cd vdo-devel
+    ```
+  * The files under src from here can be generally distinguished by language-type (though `c++` isn't accurate).
+    * Core VDO code can be found under src/c++/vdo
+    * Perl test code can be found under src/perl/Permabit
+
+For changes to the [common](https://github.com/dm-vdo/common) repository:
+
+  * Starting in the home directory, clone the repository and cd into it
+    ```
+    git clone https://github.com/dm-vdo/common.git && cd common
+    ```
+  * This layout is similar to vdo-devel, except there is no src directory at the top level.
+    * Perl code which is used for testing and other support functions can be found under perl/
+    * Packaging code to generate the various RPMs needed for Bunsen can be found under packaging/
+
+# Editing
+
+Once you've cloned your tree to the resource machine, then you can go in and start making changes.
+
+  * If you are intending to post these changes to another git repository, you will want to add the relevant repositories as a remote in the repository.
```

Comment:
"another git repository (e.g., your fork of the repository on Github)" maybe?

Reply:
Thanks.  Yes, I agree.  I will provide this in an update momentarily.

Reply:
Should be in the latest update contained within this PR.